dask_sql/physical/utils/filter.py

```diff
@@ -87,6 +95,11 @@ def attempt_predicate_pushdown(ddf: dd.DataFrame) -> dd.DataFrame:
         )
         return ddf
 
+    # Expand the filter set with provided conjunctive and disjunctive filters
+    filters.extend([f] for f in disjunctive_filters or [])
+    # Add conjunctive filters to each disjunctive filter
+    for f in filters:
+        f.extend(conjunctive_filters or [])
```

Comment:
I think this makes sense, and I think I (mostly) like the approach. However, it does seem possible that you will eventually want to do things that are currently prohibited. Some examples:

1. You want to make a disjunctive addition to your filters, and the new filter contains a conjunction that should **not** be added to any of the *other* (existing) filters. For example, you want to apply the existing filters, but you also want to allow null values for a specific user-id. (does this already work?)
2. You want to add a conjunctive filter to a subset of the pre-existing disjunctive filters.
3. You want to drop pre-existing filters.

Not sure if any of these cases are important or meaningful at all. Either way, I'm pretty sure you could support (1) and (3) later if needed.

Reply:
Thanks for the comments. You're right that all three cases aren't really support with the current addition and it only helps for passing additional filters that get applied to everything. 
Ideally I'd like to consolidate the `conjunctive` and `disjunctive` args into a single arg in dnf form which allows some flexibility with the additional filters but doesn't allow good interactivity with existing ones.

While we're unable to return a full dnf for a dataframe rather than having some existing filters coming via the hlg and some passed in additionally, I can't think of a nice way for both to interact cleanly without some good tracking around how we represent that information.